builtin/aws/ecs/releaser.go

```diff
@@ -51,107 +62,90 @@ func (r *Releaser) Release(
 	})
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failed to describe listener %q", target.ListenerArn)
+	}
+	if len(lo.Listeners) == 0 {
+		return nil, errors.Errorf("listener %q not found", target.ListenerArn)
 	}
+	listener := lo.Listeners[0]
+
+	if hostname == "" {
+
+		// We need to get the hostname from the existing alb
+		if target.LoadBalancerArn == "" {
+			s.Update("load balancer from deployment not specified - cannot determine hostname")
+		} else {
+			dlb, err := elbsrv.DescribeLoadBalancers(&elbv2.DescribeLoadBalancersInput{
+				LoadBalancerArns: []*string{aws.String(target.LoadBalancerArn)},
+			})
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to describe load balancer %q", target.LoadBalancerArn)
+			}
 
-	var listener *elbv2.Listener
+			if len(dlb.LoadBalancers) == 0 {
+				return nil, fmt.Errorf("no load balancers returned by DescribeLoadBalancers")
+			}
 
-	tgs := []*elbv2.TargetGroupTuple{
-		{
-			TargetGroupArn: &target.TargetGroupArn,
-			Weight:         aws.Int64(100),
-		},
+			hostname = *dlb.LoadBalancers[0].DNSName
+		}
 	}
 
 	log.Debug("configuring weight 100 for target group", "arn", target.TargetGroupArn)
 
-	if len(listeners.Listeners) > 0 {
-		listener = listeners.Listeners[0]
-
-		def := listener.DefaultActions
-
-		if len(def) > 0 && def[0].ForwardConfig != nil {
-			for _, tg := range def[0].ForwardConfig.TargetGroups {
-				// Drain any target groups to 0 but leave them registered.
-				// This loop also inherently removes any target groups already
-				// set to 0 that ARE NOT the one we're releasing.
-				if *tg.Weight > 0 && *tg.TargetGroupArn != target.TargetGroupArn {
-					tg.Weight = aws.Int64(0)
-					tgs = append(tgs, tg)
-					log.Debug("previous target group", "arn", *tg.TargetGroupArn)
-				}
+	def := listener.DefaultActions
+
+	if len(def) > 0 && def[0].ForwardConfig != nil {
+		for _, tg := range def[0].ForwardConfig.TargetGroups {
+			// Drain any target groups to 0 but leave them registered.
+			// This loop also inherently removes any target groups already
+			// set to 0 that ARE NOT the one we're releasing.
+			if *tg.Weight > 0 && *tg.TargetGroupArn != target.TargetGroupArn {
+				tg.Weight = aws.Int64(0)
+				tgs = append(tgs, tg)
+				log.Debug("previous target group", "arn", *tg.TargetGroupArn)
 			}
 		}
+	}
 
-		log.Debug("modifying load balancer", "tgs", len(tgs))
-		_, err = elbsrv.ModifyListener(&elbv2.ModifyListenerInput{
-			ListenerArn: listener.ListenerArn,
-			Port:        listener.Port,
-			Protocol:    listener.Protocol,
-			DefaultActions: []*elbv2.Action{
-				{
-					ForwardConfig: &elbv2.ForwardActionConfig{
-						TargetGroups: tgs,
-					},
-					Type: aws.String("forward"),
-				},
-			},
-		})
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		log.Info("load-balancer defined", "dns-name", *lb.DNSName)
-
-		lo, err := elbsrv.CreateListener(&elbv2.CreateListenerInput{
```

Comment:
I don't think this would ever happen - the deploy plugin was already creating a listener if necessary. Now we're relying on that, and only modifying the listener here.